test/processImage.js

```diff
@@ -983,7 +990,7 @@ describe('express-processimage', () => {
         }));
 
       describe('with a maxOutputPixels setting in place', () => {
-        it('should limit the size of the bounding box based on the maxOutputPixels value', () => {
+        it.skip('should limit the size of the bounding box based on the maxOutputPixels value', () => {
```

Comment:
The max pixels stuff is pretty essential to prevent gif bombs etc., so we'll need to get that working before merging.

Reply:
@papandreou limiting the pixel number is actually already working at the `impro` level no less :) these skipped tests are a case of a really particular thing supported by `express-processimage` where, if you set the maximum pixel output, it actually applies that as a factor when doing resize: https://github.com/papandreou/express-processimage/blob/master/lib/getFilterInfosAndTargetContentTypeFromQueryString.js#L892

This would need to be exposed by impro somehow, but I've not quite worked out what that would look like yet - leaning towards explicit the same way as `withoutEnlargement`.

Reply:
Hmm, yeah, that came about in https://github.com/papandreou/express-processimage/pull/27 because we don’t know the image metadata at the time where we make the decision about whether to permit the resize. This makes it tricky when one of the desired dimensions is omitted (implying a proportional resize).

The danger is then that you could bypass the max pixels limit by eg resizing a 1x10000 image with a seemingly innocent `resize=1000,`, resulting in a 1000x10000000 image 😩

I don’t consider this a feature as such. It’s not really helpful to the end user that they just end up with the largest permitted image in that case. Or maybe it could be in some cases, but it’s certainly not very predictable, especially if someone else defined the pixel limit 😅

Ideally we e-p should just reject the operation in this case. Maybe that can be arranged by impro by having it check the metadata and do the actual math?

Reply:
If Irecall correctly, only sharp and gifsicle needed this. Graphicsmagick’s pixel limiting already does (what I consider to be) the right thing.

Reply:
Ahhhh makes a ton more sense now! I've been able to take a cue from the changes in that PR and apply it to impro. All these tests now pass in e-p 😺